### PR TITLE
[2.21.x] DDF-5946 Update Intrigue to respect Subject Identity configuration

### PR DESCRIPTION
**Pull request by SmithJosh**
#### What does this PR do?
Removes all hardcoded uses of email as the subject identity. Now Intrigue uses the attribute set in the Subject Identity configuration.

#### Who is reviewing it? 
@stustison @brendan-hofmann @beyelerb 

#### How should this be tested?
1. Create two non-admin accounts in DDF
2. Change the Subject Identity config in the Admin UI to use the `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier` attribute instead of email.
3. Restart DDF
4. Verify the following:
* Metacard security attributes (the security.access-* attributes) use the name identifier instead of email.
* The workspaces page use the name identifier to indicate workspace ownership
* Workspace filtering works (Owned by Me, Not Owned by Me)
* Workspace sharing works
* Search form sharing works (top left hamburger > Search Forms)
* On load, Intrigue opens an existing workspace if there is one owned by the current user
* Query annotations use name identifier to indicate ownership, and only the owner can modify their notes.

#### What are the relevant tickets?
Fixes: #5946

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/sharing/presentation.tsx [hunk 1]
```diff
@@ -126,7 +126,7 @@ const render = (props: Props) => {
                   {item.category === Category.User ? (
                     <EditableLabelText
                       value={item.value}
-                      placeholder="user@example.com"
+                      placeholder="user id"
```
Comment (SmithJosh) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/sharing/presentation.tsx]: This is a bit unfortunate as "user id" doesn't really mean anything to the actual user

**Comment by SmithJosh** [comment]
build now

**Comment by SmithJosh** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/sharing/presentation.tsx (hunk 1 above)
Comment (brendan-hofmann) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/sharing/presentation.tsx]: ✏️ `User Id`? Like in the master PR.

**Comment by SmithJosh** [comment]
Jenkins failed to checkout PR. build now

**Comment by SmithJosh** [comment]
Same issue. I'm going to try one more time. build now

**Pull request merged**